api/v1beta3/cloudstackmachine_types.go

```diff
@@ -91,6 +99,12 @@ type CloudStackMachineSpec struct {
 	UncompressedUserData *bool `json:"uncompressedUserData,omitempty"`
 }
 
+type NetworkSpec struct {
+	Name string `json:"name"`
```

Comment:
Add `Id` so that the user can pass in the id as well. Update the id in the kubernetes spec as well.
This id will be further used while deploying the vm with additional networks.

Reply:
added in the recent commit @vishesh92 , thank you